dist-config/src/main/java/org/keycloak/config/OptionBuilder.java

```diff
@@ -0,0 +1,84 @@
+package org.keycloak.config;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+public class OptionBuilder<T> {
+    private Class<T> type;
+    private String key;
+    private OptionCategory category;
+    private Set<Option.Runtime> supportedRuntimes;
+    private boolean build;
+    private String description;
+    private Optional<T> defaultValue;
+    private List<T> expectedValues;
+
+    public OptionBuilder(String key) {
+        type = (Class<T>) String.class;
```

Comment:
I'd suggest to make it a mandatory constructor argument. Something like:
```java
public OptionBuilder(String key, Class<T> classType) {
```
This is really to avoid warnings about unsafe casting.

Reply:
I like this encoding 👍

Reply:
@andreaTP Can you please mark this as resolved?